File: ios/iLoppis/Domain/Model/VisitorTicket.swift
import Foundation

struct VisitorTicket: Identifiable, Equatable {
    let id: String
    let eventId: String
    let ticketType: String?
    let email: String?
    let status: VisitorTicketStatus
    let issuedAt: String?
    let validFrom: String?
    let validUntil: String?
    let scannedAt: String?
}

enum VisitorTicketStatus: Equatable {
    case issued
    case scanned
    case unknown(String)

    init(raw: String?) {
        switch (raw ?? "").uppercased() {
        case "SCANNED", "TICKET_STATUS_SCANNED":
            self = .scanned
        case "ISSUED", "TICKET_STATUS_NOT_SCANNED":
            self = .issued
        default:
            self = .unknown(raw ?? "")
        }
    }
}

File: android/app/src/main/java/se/iloppis/app/domain/model/VisitorTicket.kt
package se.iloppis.app.domain.model

import java.time.Instant

/**
 * Domain representation of a visitor ticket returned from the API.
 */
data class VisitorTicket(
    val id: String,
    val eventId: String,
    val ticketType: String?,
    val email: String?,
    val status: VisitorTicketStatus,
    val issuedAt: Instant?,
    val validFrom: Instant?,
    val validUntil: Instant?,
    val scannedAt: Instant?
)

/**
 * Simplified ticket status values that the UI can render.
 */
enum class VisitorTicketStatus {
    SCANNED,
    NOT_SCANNED,
    UNSPECIFIED
}
